tests/maintel/tma/test_random_walk.py

```diff
@@ -121,13 +121,13 @@ def actualPosition(self):
             )
 
             sky_offsets = []
-            async for i, az, el, off in self.script.random_walk_azel_by_time():
-                sky_offsets.append(off)
+            async for data in self.script.random_walk_azel_by_time():
```

Comment:
Way up above in test_run, about line 70 I see you use a mock for random_walk_azel_by_time. That explains why test_run passed even when you broke the script. You also mock slew_and_track, which means this test is testing essentially none of your code.

I suggest mocking the "tcs" instead, if practical, though that will clearly be more work. (I also asked Tiago to consider adding a simulation mode to these classes and copied you on the ticket.)

Reply:
I **think** I covered what you suggested.